src/mpcCoreKit.ts

```diff
@@ -935,23 +1048,43 @@ export class Web3AuthMPCCoreKit implements ICoreKit {
       } else {
         await this.addFactorDescription(factorKey, FactorKeyTypeShareDescription.HashedShare);
       }
+    });
+  }
 
-      // workaround for atomic sync, commit changes if not manualSync
-      if (!this.options.manualSync) await this.commitChanges();
+  protected async atomicSync<T>(f: () => Promise<T>): Promise<T> {
+    this.tkey.manualSync = true;
+    try {
+      const r = await f();
+      await this.commitChanges();
+      return r;
+    } finally {
+      this.tkey.manualSync = this.options.manualSync;
+    }
+  }
+
+  private async handleExistingUser() {
+    await this.tKey.initialize({ neverInitializeNewKey: true });
+    if (this.options.disableHashedFactorKey) {
+      return;
+    }
+
+    const hashedFactorKey = getHashedPrivateKey(this.state.oAuthKey, this.options.hashedFactorNonce);
+    this.state.factorKey = hashedFactorKey;
+    if (await this.checkIfFactorKeyValid(hashedFactorKey)) {
```

Comment:
I noticed there was previously an unexpected behavior regarding initialization of an existing user whose key hashedFactorKey storage was previously deleted.

In this case, the login would not throw an error, but the factor key would not be set and tkey initialization would not have been finalized.

I've changed this behavior to always set the factor key (this is needed in tests to reset account), and warn if share has been deleted.

I was also thinking whether we should throw an error in this case. What do you think? @ieow @himanshuchawla009